actionview/lib/action_view/helpers/url_helper.rb

```diff
@@ -379,48 +380,52 @@ def link_to_unless_current(name, options = {}, html_options = {}, &block)
       end
 
       # Creates a link tag of the given +name+ using a URL created by the set of
-      # +options+ unless +condition+ is true, in which case only the name is
-      # returned. To specialize the default behavior (i.e., show a login link rather
-      # than just the plaintext link text), you can pass a block that
-      # accepts the name or the full argument list for +link_to_unless+.
+      # +options+, unless +condition+ is true. Otherwise only the name is returned or
+      # the given block is yielded. Therefore you can use the block to specialize
+      # the default behaviour, e.g. change the link target (example below).
+      # This behaviour is different from how the block works for +link_to+.
       #
       # ==== Examples
       #   <%= link_to_unless(@current_user.nil?, "Reply", { action: "reply" }) %>
       #   # If the user is logged in...
       #   # => <a href="/controller/reply/">Reply</a>
+      #   # If not...
+      #   # => Reply
       #
       #   <%=
       #      link_to_unless(@current_user.nil?, "Reply", { action: "reply" }) do |name|
       #        link_to(name, { controller: "accounts", action: "signup" })
       #      end
       #   %>
       #   # If the user is logged in...
-      #   # => <a href="/controller/reply/">Reply</a>
+      #   # => <a href="/controller/reply">Reply</a>
       #   # If not...
       #   # => <a href="/accounts/signup">Reply</a>
       def link_to_unless(condition, name, options = {}, html_options = {}, &block)
         link_to_if !condition, name, options, html_options, &block
       end
 
       # Creates a link tag of the given +name+ using a URL created by the set of
-      # +options+ if +condition+ is true, otherwise only the name is
-      # returned. To specialize the default behavior, you can pass a block that
-      # accepts the name or the full argument list for +link_to_unless+ (see the examples
-      # in +link_to_unless+).
+      # +options+, if +condition+ is true. Otherwise only the name is returned or
+      # the given block is yielded. Therefore you can use the block to specialize
+      # the default behaviour, e.g. change the link target (example below).
+      # This behaviour is different from how the block works for +link_to+.
```

Comment:
This one isn't awkward, but I'd still rephrase as above -- mostly, opening with a conditional makes it clear that different things are happening -- with it buried at the end of the sentence, a first glance is going to imply that this method always "Creates a link tag".